Source/MeleeTrace/Private/MeleeTraceComponent.cpp

```diff
@@ -278,21 +283,25 @@ void UMeleeTraceComponent::InternalEndTrace(uint32 TraceHash)
 		return;
 	}
 
-	const int32 FoundIndex = ActiveMeleeTraces.IndexOfByPredicate(
-		[TraceHash](const FActiveMeleeTraceInfo& ActiveMeleeTraceInfo)
-		{
-			return TraceHash == ActiveMeleeTraceInfo.TraceHandle.TraceHash;
-		});
+	auto Predicate = [TraceHash](const FActiveMeleeTraceInfo& ActiveMeleeTraceInfo)
+	{
+		return TraceHash == ActiveMeleeTraceInfo.TraceHandle.TraceHash;
+	};
+	int32 FoundIndex = ActiveMeleeTraces.IndexOfByPredicate(Predicate);
 
 	if (ensureAlwaysMsgf(FoundIndex != INDEX_NONE,
 		TEXT("Attemping to end trace with hash: %u but no trace with hash exist"),
 		TraceHash))
 	{
-		OnTraceEnd.Broadcast(
-			this,
-			ActiveMeleeTraces[FoundIndex].HitActors.Num(),
-			ActiveMeleeTraces[FoundIndex].TraceHandle);
-		ActiveMeleeTraces.RemoveAtSwap(FoundIndex);
+		while (FoundIndex != INDEX_NONE)
+		{
+			OnTraceEnd.Broadcast(
+				this,
+				ActiveMeleeTraces[FoundIndex].HitActors.Num(),
+				ActiveMeleeTraces[FoundIndex].TraceHandle);
+			ActiveMeleeTraces.RemoveAtSwap(FoundIndex);
+			FoundIndex = ActiveMeleeTraces.IndexOfByPredicate(Predicate);
+		}
```

Comment:
This isn't  really necessary. Each trace should have unique hash. Did you add this code just as a caution or did you actually run into a case where there were two traces active at the same time with identical hash?

Reply:
@rlewicki I ran into the case.

In this PR I'm enabling adding multiple traces, which are added in `InternalStartTrace` - where we already have the hash. Hence all the traces added in there will have the same hash.

The issue without this part is that only the 1st trace would be removed; e.g. for 2 swords, the 2nd is left and kept tracing.


Probably the more "correct" code would be to make the hash unique by both the context object and the mesh object.
But honestly there's just no need; simply considering this hash is now a "group hash", and removing them all in this loop, works perfectly.